gym/spaces/discrete.py

```diff
@@ -40,14 +40,35 @@ def __init__(
         self.start = int(start)
         super().__init__((), np.int64, seed)
 
-    def sample(self) -> int:
+    def sample(self, mask: Optional[np.ndarray] = None) -> int:
         """Generates a single random sample from this space.
 
-        A sample will be chosen uniformly at random.
+        A sample will be chosen uniformly at random with the mask if provided
+
+        Args:
+            mask: An optional mask for if an action can be selected. Expected shape is (n,). If not possible actions, will default to `space.start`
```

Comment:
We should specify that `mask` should be a binary array of dtype `np.int8`. We should also say whether `mask` specifies the legal or illegal samples. I would avoid references to actions, since this is about spaces in general.

I'm not sure what you mean by "will default to `space.start`, but I guess you mean that the *sample* will default to that, right?

Reply:
Updated, could you confirm